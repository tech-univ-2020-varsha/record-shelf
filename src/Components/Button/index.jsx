import React, { useState } from 'react';
import propTypes from 'prop-types';
import * as styles from './index.module.css';

const Button = ({
  testId, children, buttonOnClick,
}) => (
  <button className={styles.buttonStyle} type="submit" data-testid={testId} onClick={buttonOnClick}>{children}</button>
);

export default Button;

Button.propTypes = {
  testId: propTypes.string,
  children: propTypes.string.isRequired,
  buttonOnClick: propTypes.func.isRequired,
};
