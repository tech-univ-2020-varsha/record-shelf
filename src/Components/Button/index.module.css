.buttonStyle{
    background-color: rgb(93, 209, 51);
    border-radius: 10%;
    padding-left: 2em;
    padding-right: 2em;
    padding-top: 1em;
    padding-bottom: 1em;
    color: white;
    font-family:Poppins;
    font-weight: bold;
    font-size: large;
    height: 30%;
    width: 60%;
}