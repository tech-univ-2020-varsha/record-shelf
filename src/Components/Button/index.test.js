import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './index';

describe('the button component', () => {
  it('should render the button corretly', () => {
    const { asFragment } = render(<Button testId="test-btn">Test text</Button>);

    expect(asFragment()).toMatchSnapshot();
  });
  it('should call the onClick prop when the button is clicked', () => {
    const onClickMock = jest.fn();
    const { getByTestId } = render(<Button testId="test-btn" buttonOnClick={onClickMock}>test text</Button>);
    fireEvent.click(getByTestId('test-btn'));
    expect(onClickMock).toHaveBeenCalled();
  });
});
