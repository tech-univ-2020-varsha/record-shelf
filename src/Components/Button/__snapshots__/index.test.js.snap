// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button component should render the button corretly 1`] = `
<DocumentFragment>
  <button
    data-testid="test-btn"
    type="submit"
  >
    Test text
  </button>
</DocumentFragment>
`;
